### PR TITLE
fix: Handle unsanitized markup in DOM

### PR DESCRIPTION
**Pull request by fayazara**
## Description
Sanitizes all message before they are rendered on the DOM

Before

https://user-images.githubusercontent.com/15716057/156751583-9e5dc61e-c30e-4c29-9812-49fbc356a7ce.mp4

After

https://user-images.githubusercontent.com/15716057/156751942-4be52f17-0b70-463c-91d8-ca7b291cb923.mp4

DOMpurify will strip all unnecessary markup before it gets passed to remdered.

Browsers tested 
1. Chrome
2. Safari
3. Firefox

Fixes [#294](https://github.com/chatwoot/product/issues/294)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

1. Tested on Chat bubbles with Normal link and javascript alerts
2. Tested on private notes
3. Tested on contact details pane

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by fayazara** [comment]
Using DOMPurify Sanitize method to clean the markup

Example - https://cure53.de/purify
CleanShot 2022-03-04 at 16 14 37

File: app/javascript/shared/mixins/messageFormatterMixin.js [hunk 1]
```diff
@@ -5,7 +5,12 @@ export default {
   methods: {
     formatMessage(message, isATweet) {
       const messageFormatter = new MessageFormatter(message, isATweet);
-      return messageFormatter.formattedMessage;
+      DOMPurify.addHook('afterSanitizeAttributes', function(node) {
+        if ('target' in node) {
+          node.setAttribute('target', '_blank');
+        }
+      });
+      return DOMPurify.sanitize(messageFormatter.formattedMessage);
```
Comment (pranavrajs) [inline app/javascript/shared/mixins/messageFormatterMixin.js:13]: @fayazara 

Can we do this inside `formattedMessage` getter in MessageFormatter?

Comment (fayazara) [reply app/javascript/shared/mixins/messageFormatterMixin.js]: Resolved in https://github.com/chatwoot/chatwoot/pull/4110/commits/77f0df3af58f42093a617d37977e42c2cd48cc64

**Comment by pranavrajs** [comment]
@fayazara Can you please add the specs to cover the case you handled in the PR?

**Comment by fayazara** [comment]
> @fayazara Can you please add the specs to cover the case you handled in the PR?

Added in https://github.com/chatwoot/chatwoot/pull/4110/commits/a6509ea22c4881c8f23349fbe1d7acf0e398a401

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**